### PR TITLE
refactor(build): Pre-compile files imported from data folder

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

According to microsoft/TypeScript#10939, it would be tricky to just `import example from 'example.txt''`.
Thus, we're using `data/` folder for this.
After this PR, we pre-compile files from this folder and import it as JavaScript `Map`.
[Generated file example](https://gist.github.com/zharinov/d83a2c1379db066538c276216771a18d)

## Context:

Ref: #7157

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

File: tools/generate-imports.ts [hunk 1]
```diff
@@ -64,9 +70,44 @@ async function generate({
...
+
+  const importDataFileType = files.map((x) => `  | '${x}'`).join('\n');
+
+  const contentMapDecl = 'const data = new Map<DataFile, string>();';
+
+  const contentMapAssignments = [];
+  for (const file of files) {
+    shell.echo(`> data/${file}`);
+    const rawFileContent = await fs.readFile(`data/${file}`, 'utf8');
+    const prettyFileContent = `\n${rawFileContent.replace(/`/g, '\\`')}\n`;
```
Comment (viceice) [inline tools/generate-imports.ts:88]: I think `JSON.stringify` is a more secure solution

File: tools/generate-imports.ts [hunk 2]
```diff
@@ -64,9 +70,43 @@ async function generate({
...
+    .readdirSync('data', { withFileTypes: true })
+    .filter((dirent) => dirent.isFile())
+    .map((dirent) => dirent.name)
+    .sort();
+
+  const importDataFileType = files.map((x) => `  | '${x}'`).join('\n');
+
+  const contentMapDecl = 'const data = new Map<DataFile, string>();';
+
+  const contentMapAssignments = [];
```
Comment (viceice) [inline tools/generate-imports.ts:84]:
```suggestion
  const contentMapAssignments: string[] = [];
```

**Review by viceice — CHANGES_REQUESTED** [review]
Looks like it need coverage fixes now 🙃

**Comment by zharinov** [comment]
Ha, didn't expect it will help to get rid of dependency I found problematic some time ago

File: yarn.lock [hunk 3]
```diff
@@ -7807,14 +7790,8 @@ npm@^6.14.8:
     libnpx "^10.2.4"
     lock-verify "^2.1.0"
     lockfile "^1.0.4"
-    lodash._baseindexof "*"
     lodash._baseuniq "~4.6.0"
-    lodash._bindcallback "*"
-    lodash._cacheindexof "*"
-    lodash._createcache "*"
-    lodash._getnative "*"
```
Comment (viceice) [inline yarn.lock:7815]: haha, renovate yarn patch strikes again 😛

**Review by viceice — DISMISSED** [review]

File: yarn.lock (hunk 3 above)
Comment (zharinov) [reply yarn.lock:7815]: I was wrong about `find-up`

**Comment by viceice** [comment]
🤦‍♂️  nuget needs find-up
image

**Review by viceice — APPROVED** [review]

**Pull request merged**